snuba/query/snql/__init__.py

```diff
@@ -1,28 +1,255 @@
+    ) -> Expression:
+        return visit_high_pri_arithmetic(node, visited_children)
+
+    def visit_numeric_literal(
+        self, node: Node, visited_children: Iterable[Any]
+    ) -> Literal:
+        return visit_numeric_literal(node, visited_children)
+
+    def visit_quoted_literal(
+        self, node: Node, visited_children: Tuple[Any, Node, Any]
+    ) -> Literal:
+        return visit_quoted_literal(node, visited_children)
+
+    def visit_order_by_clause(self, node, visited_children):
+        _, _, order_columns, _ = visited_children
+        return order_columns
+
+    # order_columns* (low_pri_arithmetic (("ASC"/"DESC"))
+    def visit_order_list(self, node, visited_children):
+        left_order_list, right_order, order = visited_children
+
+        ret: List[OrderBy] = []
+        if not isinstance(left_order_list, Node):
+            if not isinstance(left_order_list, (list, tuple)):
+                ret.append(left_order_list)
+            else:
+                for p in left_order_list:
+                    ret.append(p)
+
+        if order.text == "ASC":
+            ret.append(OrderBy(OrderByDirection.ASC, right_order))
+        elif order.text == "DESC":
+            ret.append(OrderBy(OrderByDirection.DESC, right_order))
+
+        return ret
+
+    def visit_order_columns(self, node, visited_children):
+        column, order, _, _, _ = visited_children
+
+        if order.text == "ASC":
+            order_object = OrderBy(OrderByDirection.ASC, column)
+        elif order.text == "DESC":
+            order_object = OrderBy(OrderByDirection.DESC, column)
+
+        return order_object
+
+    def visit_group_by_clause(self, node, visited_children):
+        _, _, group_columns, _ = visited_children
+        return group_columns
+
+    def visit_group_list(self, node, visited_children):
+        left_group_list, right_group = visited_children
+        ret: List[Expression] = []
+        if not isinstance(left_group_list, Node):
+            if not isinstance(left_group_list, (list, tuple)):
+                ret.append(left_group_list)
+            else:
+                for p in left_group_list:
+                    ret.append(p)
+
+        ret.append(right_group)
+        return ret
+
+    def visit_collect_clause(self, node, visited_children):
+        _, _, selected_columns, _ = visited_children
+        return selected_columns
+
+    def visit_collect_columns(self, node, visited_children):
+        columns, _, _, _, = visited_children
+        return columns
+
+    def visit_collect_list(self, node, visited_children):
+        raw_list = node.text.split(",")
+        i = 0
+        column_list, right_column = visited_children
+        ret: List[SelectedExpression] = []
+        if not isinstance(column_list, Node):
+            if not isinstance(column_list, (list, tuple)):
+                if column_list.alias is None:
+                    ret.append(SelectedExpression(node.text, column_list))
```

Comment:
I am not sure node.text is the right way to extract the raw string here. It seems to me that it would not work if the collect clause contains only two expressions. We would ennd here but node.text would contain the whole raw collect expression. 

The safe way of doing this is by returning the raw string together with the Expression object from `visit_collect_clause`. Then you just need to pick the raw string provided there.